.github/workflows/continuous-integration.yml

```diff
@@ -1,150 +1,120 @@
 name: Continuous Integration
-on: 
-  push: 
-    branches-ignore:    
-      - documentation
-  pull_request: 
-    branches-ignore:    
-      - documentation
+on:
+  push:
+    branches-ignore:
+    - documentation
+  pull_request:
+    branches-ignore:
+    - documentation
+
+defaults:
+  run:
+    # Enable Conda environment by using the login shell:
+    shell: bash -leo pipefail {0}
 
 jobs:
   CI:
     runs-on: ubuntu-20.04
-    defaults:
-      run:
-        shell: bash -l {0}
     strategy:
+      fail-fast: false
       matrix:
-        fortran-compiler:  [gfortran-9, gfortran-10]
+        fortran-compiler: [gfortran-9, gfortran-10]
         rte-kernels: [default, openacc]
+        include:
+        - fcflags: "-ffree-line-length-none -m64 -std=f2008 -march=native -fbounds-check -finit-real=nan -g -DRTE_USE_CBOOL"
+        - netcdf-fortran: /usr
     env:
+      # Core variables:
       FC: ${{ matrix.fortran-compiler }}
-      FCFLAGS: "-ffree-line-length-none -m64 -std=f2008 -march=native -fbounds-check -finit-real=nan -g -DRTE_USE_CBOOL"
-      CC: gcc
-      NCHOME: /home/runner/netcdf-c
-      NFHOME: /home/runner/netcdf-fortran
-      RFMIP_DIR: /home/runner/rfmip-files
+      FCFLAGS: ${{ matrix.fcflags }}
+      # Make variables:
+      NCHOME: /dummy
+      NFHOME: ${{ matrix.netcdf-fortran }}
+      RRTMGP_ROOT: ${{ github.workspace }}
+      RTE_KERNELS: ${{ matrix.rte-kernels }}
+      RUN_CMD:
+      # Auxiliary variables:
+      RFMIP_CACHEDIR: /home/runner/rfmip-files
     steps:
-    - name: Update system packages
+    #
+    # Checks-out repository under $GITHUB_WORKSPACE
+    #
+    - uses: actions/checkout@v3
+    #
+    # Synchronize the package index
+    #
+    - name: Synchronize the package index
       run: sudo apt-get update
-    ############################################################################
     #
-    # Compilers....
+    # Install Gfortran 10
     #
-    # Gfortran 10 not available in Github CI stack, so install
+    - name: Install Gfortran 10
+      if: matrix.fortran-compiler == 'gfortran-10'
+      run: sudo apt-get install gfortran-10
     #
-    - name: gfortran-10 setup compiler
-      if: contains(matrix.fortran-compiler, 'gfortran-10')
-      run: |
-        sudo apt-get install gfortran-10 gcc-10
-        echo "CC=gcc-10" >> $GITHUB_ENV
-
-    ############################################################################
+    # Install NetCDF-Fortran (compatible with both compilers)
     #
-    # Netcdf C and Fortran
+    - name: Install NetCDF-Fortran
+      run: sudo apt-get install libnetcdff-dev
```

Comment:
@skosukhin I love this idea. Will it also work if we move to Ubuntu 22.04 and use gfortran 10, 11, 12? Should I install netcdff-dev this way in the containers for uniformity?

Reply:
@skosukhin Scratch the suggestion about containers - of course netCDF Fortran has to be built with the Fortran compiler we're testing.